### PR TITLE
Add GenIso.fields

### PR DESCRIPTION
**Pull request by japgolly**
Addresses #295.

Included is a benchmark for comparison with `productToTuple`.
I also took the liberty of upgrading JMH from 0.2.4 to 0.2.5 while I was there. Hope that's ok.

### CODE REVIEW

**Comment by japgolly** [comment]
Benchmark results are in. It's looks to be around 2x ~ 4x faster on average.

```
Benchmark                            (pIsos)  (pSubject)  Mode  Cnt     Score     Error   Units
ProductIsoBench.get                   fields       mono2  avgt   25    22.314 ±   0.115   ns/op
ProductIsoBench.get                shapeless       mono2  avgt   25    84.764 ±   0.681   ns/op
ProductIsoBench.get                   fields       mono4  avgt   25    32.396 ±   0.644   ns/op
ProductIsoBench.get                shapeless       mono4  avgt   25   144.348 ±   0.753   ns/op
ProductIsoBench.get                   fields       mono8  avgt   25    48.623 ±   0.275   ns/op
ProductIsoBench.get                shapeless       mono8  avgt   25   155.616 ±   6.728   ns/op
ProductIsoBench.get                   fields       poly2  avgt   25    18.467 ±   0.108   ns/op
ProductIsoBench.get                shapeless       poly2  avgt   25    81.660 ±   0.292   ns/op
ProductIsoBench.get                   fields       poly4  avgt   25    23.064 ±   1.233   ns/op
ProductIsoBench.get                shapeless       poly4  avgt   25   130.844 ±   2.786   ns/op
ProductIsoBench.get                   fields       poly8  avgt   25    35.786 ±   0.103   ns/op
ProductIsoBench.get                shapeless       poly8  avgt   25   152.151 ±   0.463   ns/op
ProductIsoBench.reverseGet            fields       mono2  avgt   25    24.791 ±   0.590   ns/op
ProductIsoBench.reverseGet         shapeless       mono2  avgt   25    96.771 ±   1.465   ns/op
ProductIsoBench.reverseGet            fields       mono4  avgt   25    38.037 ±   0.237   ns/op
ProductIsoBench.reverseGet         shapeless       mono4  avgt   25    88.633 ±   0.413   ns/op
ProductIsoBench.reverseGet            fields       mono8  avgt   25    58.138 ±   0.347   ns/op
ProductIsoBench.reverseGet         shapeless       mono8  avgt   25   165.930 ±   0.450   ns/op
ProductIsoBench.reverseGet            fields       poly2  avgt   25    27.232 ±   3.852   ns/op
ProductIsoBench.reverseGet         shapeless       poly2  avgt   25    99.029 ±   0.799   ns/op
ProductIsoBench.reverseGet            fields       poly4  avgt   25    30.926 ±   1.935   ns/op
ProductIsoBench.reverseGet         shapeless       poly4  avgt   25   170.004 ±   1.840   ns/op
ProductIsoBench.reverseGet            fields       poly8  avgt   25    44.156 ±   0.111   ns/op
ProductIsoBench.reverseGet         shapeless       poly8  avgt   25   169.203 ±   7.615   ns/op
```

File: project/plugins.sbt [hunk 1]
```diff
@@ -1,3 +1,5 @@
+addSbtPlugin("org.scala-js" % "sbt-scalajs" % "0.6.5")
```
Comment (julien-truffaut) [inline project/plugins.sbt]: why do you need scalajs?

File: macro/src/main/scala/monocle/macros/GenIso.scala [hunk 2]
```diff
@@ -59,27 +71,86 @@ class GenIsoImpl(val c: blackbox.Context) {
...
       caseAccessorsOf[S] match {
         case Nil =>
           val sTpeSym = sTpe.typeSymbol.companion
-          c.Expr[Iso[S, Unit]](q"""
+          q"""
             monocle.Iso[${sTpe}, Unit](Function.const(()))(Function.const(${sTpeSym}()))
-          """)
-        case _   => c.abort(c.enclosingPosition, s"$sTpe needs to be a case class with no accessor or an object.")
+          """
+        case _   => fail(s"$sTpe needs to be a case class with no accessor or an object.")
```
Comment (julien-truffaut) [inline macro/src/main/scala/monocle/macros/GenIso.scala]: what's the difference between `fail` and `c.abort`?

**Comment by julien-truffaut** [comment]
It looks great but there seems to be an issue with scala 2.10

File: macro/src/main/scala/monocle/macros/GenIso.scala (hunk 2 above)
Comment (NightRa) [inline macro/src/main/scala/monocle/macros/GenIso.scala]: He defined `fail` to be short of `abort`

Comment (julien-truffaut) [inline macro/src/main/scala/monocle/macros/GenIso.scala]: thanks I missed that

Comment (japgolly) [inline macro/src/main/scala/monocle/macros/GenIso.scala]: Yeah, I find it helps readability in contrast to `c.abort(c.enclosingPosition, _)` all over. Hope you don't mind :)

File: project/plugins.sbt (hunk 1 above)
Comment (japgolly) [inline project/plugins.sbt]: Oh! That's a mistake; I'll remove that.

**Comment by japgolly** [comment]
Ok, I didn't even realise that Scala 2.10 was choking and vomiting all over itself with the new macro. Lucky for TravisCI, thanks for pointing that out to me.

So, I've made the macros work with Scala 2.10, Travis is green and we should be good now. For reference I also opened a macro-compat ticket: https://github.com/milessabin/macro-compat/issues/25

**Comment by julien-truffaut** [comment]
Thank you very much! :+1:

**Pull request merged**